filters/diag/diag.go

```diff
@@ -214,30 +213,14 @@ func (r *random) Response(ctx filters.FilterContext) {}
 func (r *repeat) Name() string { return RepeatName }
 
 func (r *repeat) CreateFilter(args []interface{}) (filters.Filter, error) {
-	if len(args) != 2 {
-		return nil, filters.ErrInvalidFilterParameters
-	}
-
-	text, ok := args[0].(string)
-	if !ok || text == "" {
-		return nil, filters.ErrInvalidFilterParameters
-	}
-
-	var len int64
-	switch v := args[1].(type) {
-	case float64:
-		len = int64(v)
-	case int:
-		len = int64(v)
-	case int64:
-		len = v
-	default:
-		return nil, filters.ErrInvalidFilterParameters
+	a := filters.Args(args)
+	text, len, err := a.String(), a.Int64(), a.Err()
+	if err != nil {
+		return nil, err
 	}
-	if len < 0 {
+	if text == "" || len < 0 {
```

Comment:
The converter could be extended to provide common checks like:
```go
a := filters.Args(args)
return &repeat{[]byte(a.StringNonEmpty()), a.Int64GreaterEqual(0)}, a.Err()
```

Reply:
It is possible to add validating methods but their number could quickly go out of hands. It is also not possible to cover all validation usecases anyway because validation may require cross-argument check and additional logic (e.g. library call).
Therefore the good balance seems to be to leave type conversion to the converter and implement validation on the caller side.